libs/shared/storage/src/lib/shared-storage.service.ts

```diff
@@ -126,19 +127,24 @@ export class SharedStorageService {
   }
 
   public async putObject(
-    id: string,
+    key: string,
     data: Buffer | Readable,
     isPrivate = false,
   ): Promise<AWS.S3.ManagedUpload.SendData> {
     const acl = isPrivate ? 'private' : 'public-read';
     const request: AWS.S3.PutObjectRequest = {
-      Key: id,
-      Body: data,
+      Key: key,
+      Body: isReadableStream(data) 
+        ? new PassThrough() 
+        : data,
       Bucket: this.BUCKET,
       ACL: acl,
     };
 
     const call = this.s3.upload(request).promise();
+    if (isReadableStream(data)) {      
+      (data as Readable).pipe(request.Body as PassThrough);
```

Comment:
Are these casts needed? If so, then the typegaurd doesn't seem to working, which is confusing since it seems correct to me

Reply:
That didn't work for me. The casts were needed.